soc: mcxa156: use soc hooks

nashif opened the pull request:
Use SoC hooks instead of legacy z_arm_platform_init.

Fixes #78386

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
soc: mcxa156: use soc hooks

Use SoC hooks instead of legacy z_arm_platform_init.

Fixes #78386

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

gmarull committed:
arch: arm: do not enable PLATFORM_SPECIFIC_INIT if SOC_RESET_HOOK=y

Otherwise we can't escape from DEPRECATED being selected, and so getting
build warnings. It doesn't make sense that the option replacing the
deprecated one is used to automatically enable it.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull commented on soc/nxp/mcx/mcxa/soc.c:
```diff
@@ -17,9 +17,9 @@
 #include <soc.h>
 
-#ifdef CONFIG_PLATFORM_SPECIFIC_INIT
+#ifdef CONFIG_SOC_RESET_HOOK
```
ifdef is useless, can be removed

nashif committed:
manifest: update to sha with fix to PLATFORM_SPECIFIC_INIT

PLATFORM_SPECIFIC_INIT is deprecated.

Update nxp hal to sha 77815705c465627b8436cbac51f0bf0594bbeba2
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
@mmahadevan108  @danieldegrasse  please review this one

danieldegrasse commented on soc/nxp/mcx/mcxa/soc.c:
```diff
@@ -17,11 +17,7 @@
 #include <zephyr/init.h>
 #include <soc.h>
 
-#ifdef CONFIG_PLATFORM_SPECIFIC_INIT
```
I believe the `ifdef` here is not useless- we don't compile in the code implementing `SystemInit` unless `CONFIG_PLATFORM_SPECIFIC_INIT` is set: https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/hal_nxp.cmake#L101

So we might need a change to the HAL as well there, to use `CONFIG_SOC_RESET_HOOK`?

nashif replied:
ok, re-added it to be consistent with other usages, can be removed later everywhere if we decide it is not needed.

gmarull replied:
Doesn’t the soc always select the option?
